viadot/tasks/epicor.py

```diff
@@ -0,0 +1,239 @@
+    """
+    Function to parse xml containing Epicor Orders Data.
+
+    Args:
+        xml_data (str, required): Response from Epicor API in form of xml
+    Returns:
+        pd.DataFrame: DataFrame containing parsed orders data.
+
+    """
+    full_df = pd.DataFrame()
+    ship_dict = {}
+    invoice_dict = {}
+    header_params_dict = {}
+    item_params_dict = {}
+    almost_full = {}
+
+    root = ET.fromstring(xml_data.text)
+    for order in root.findall("Order"):
+        for header in order.findall("HeaderInformation"):
+            for tracking_numbers in header.findall("TrackingNumbers"):
+                numbers = ""
+                for tracking_number in tracking_numbers.findall("TrackingNumber"):
+                    numbers = numbers + "'" + tracking_number.text + "'"
+                result_numbers = TrackingNumbers(TrackingNumber=numbers)
+                almost_full.update(result_numbers)
+
+            for shipto in header.findall("ShipToAddress"):
+                for ship_param in ShipToAddress.__dict__.get("__annotations__"):
+                    try:
+                        ship_value = shipto.find(f"{ship_param}").text
+                    except:
+                        ship_value = None
+                    ship_parameter = {ship_param: ship_value}
+                    ship_dict.update(ship_parameter)
+                almost_full.update(ship_dict)
+
+            for invoice in header.findall("InvoiceTotals"):
+                for invoice_param in InvoiceTotals.__dict__.get("__annotations__"):
+                    try:
+                        invoice_value = invoice.find(f"{invoice_param}").text
+                    except:
+                        invoice_value = None
+                    invoice_parameter = {invoice_param: invoice_value}
+                    invoice_dict.update(invoice_parameter)
+                almost_full.update(invoice_dict)
+
+            for header_param in HeaderInformation.__dict__.get("__annotations__"):
+                try:
+                    header_value = header.find(f"{header_param}").text
+                except:
+                    header_value = None
+                header_parameter = {header_param: header_value}
+                header_params_dict.update(header_parameter)
+            almost_full.update(header_params_dict)
+
+        for items in order.findall("LineItemDetails"):
+            for item in items.findall("LineItemDetail"):
+                for item_param in LineItemDetail.__dict__.get("__annotations__"):
+                    try:
+                        item_value = item.find(f"{item_param}").text
+                    except:
+                        item_value = None
+                    item_parameter = {item_param: item_value}
+                    item_params_dict.update(item_parameter)
+                almost_full.update(item_params_dict)
+                full_df = full_df.append(almost_full, ignore_index=True)
+    return full_df
```

Comment:
1) All of that should be in the source; the task should simply call source.to_df().
2) You define the models but don't use them in your parsing function?
3) Are all those fields really optional? If yes, pls add a note saying that eg. the official documentation does not specify the list of required fields and this is why we made them all optional. If not, you shouldn't specify them as optional as then you won't know when a required field is missing.